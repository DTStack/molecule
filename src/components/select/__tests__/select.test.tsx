import React from 'react';
// import { fireEvent, screen, render } from '@testing-library/react';
import renderer from 'react-test-renderer';

import { Select, Option } from '../index';

function SelectTest(props) {
    return (
        <Select
            id="demo1"
            key="demo1"
            defaultValue="1"
            style={{
                width: 200,
                color: 'rgba(255, 255, 255, 0.4)',
                background: '#252526',
            }}
            {...props}
        >
            <Option value="1">option - 1</Option>
            <Option value="2">option - 2</Option>
            <Option value="3">option - 3</Option>
            <Option value="4" description="Test option one">
                option - 4
            </Option>
        </Select>
    );
}

describe('Test Select Component', () => {
    // Clean up the html manually
    afterEach(() => {
        document.body.innerHTML = '';
    });

    test('Match the Select snapshot', () => {
        const component = renderer.create(<SelectTest />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('Listen to the onSelect event', () => {
        // const body = render(<SelectTest />);
        // expect(tree).toMatchSnapshot();
    });
});
