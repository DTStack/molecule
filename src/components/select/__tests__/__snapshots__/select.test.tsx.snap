// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Select Component Match the Select snapshot 1`] = `
<div
  className="mo-select"
  defaultValue="1"
  id="demo1"
  style={
    Object {
      "background": "#252526",
      "color": "rgba(255, 255, 255, 0.4)",
      "width": 200,
    }
  }
>
  <input
    autoComplete="off"
    className="mo-select__input"
    onClick={[Function]}
    readOnly={true}
    value="option - 1"
  />
  <span
    className="mo-select__arrow"
  >
    <span
      className="codicon codicon-chevron-down"
    />
  </span>
</div>
`;
